src/client/reseed.cc

```diff
@@ -237,22 +238,31 @@ bool Reseed::FetchStream(
  *   - Extract RI's for Reseed
  *
  */
-bool SU3::SU3Impl() {
-  /**
-   * TODO(unassigned): when the --reseed-from file is a .zip, (or non-su3 type),
-   * it would be nice to skip su3 all-together, decompress as appropriate, and
-   * validate for RI type - then insert into NetDb.
-   */
-  LOG(debug) << "SU3: preparing stream...";
-  if (!PrepareStream()) {
-    LOG(error) << "SU3: preparation failed";
-    return false;
-  }
-  LOG(debug) << "SU3: verifying stream...";
-  if (!VerifySignature()) {
-    LOG(error) << "SU3: verification failed";
-    return false;
-  }
+bool SU3::SU3Impl()
+{
+  if (boost::algorithm::ends_with(kovri::context.GetOptionReseedFrom(), ".zip"))
```

Comment:
added option --disable-su3-verification. i think with this option, an early detection is not needed anymore since it's the first thing done in ZIP impl and we only support zip, so the check will be redundant. Or maybe i should add a todo with "detection an implementation of other format" ?